compiler/luci/logex/src/CircleNodeSummaryBuilder.cpp

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "CircleNodeSummaryBuilder.h"
+#include "CircleNodeSummaryBuilders.h"
+
+#include <memory>
+
+namespace luci
+{
+
+std::shared_ptr<CircleNodeSummaryBuilder>
+CircleNodeSummaryBuilder::create_builder(const luci::CircleNode *node)
+{
+  switch (node->opcode())
+  {
+#define CIRCLE_NODE(OPCODE, CLASS)    \
+  case luci::CircleOpcode::OPCODE:    \
+  {                                   \
+    return std::make_shared<CLASS>(); \
+  }
```

Comment:
```suggestion
#define CIRCLE_NODE(OPCODE, CLASS)    \
  case luci::CircleOpcode::OPCODE: return std::make_shared<CLASS>();
```
I'm not sure how line count works but this seems to reduce line count of the function. If possible could you check this too?

Reply:
This is counted as one line :)
So LoC of `CircleNodeSummaryBuilder` is 192!

Reply:
both are 192 same ?

Reply:
The number of `CIRCLE_NODE(...)` is 133 and each `CIRCLE_NODE(...)` is counted as 1.
As a result, LoC of `CircleNodeSummaryBuilder` is 192 because `CircleNodeSummaryBuilder` includes other member functions such as `validate(const luci::CircleNode *node)`, `update_status(locop::NodeSummary &s)`, etc :)

Reply:
I mean
```
#define CIRCLE_NODE(OPCODE, CLASS)    \
  case luci::CircleOpcode::OPCODE:    \
  {                                   \
    return std::make_shared<CLASS>(); \
  }
```
vs
```
#define CIRCLE_NODE(OPCODE, CLASS)    \
  case luci::CircleOpcode::OPCODE: return std::make_shared<CLASS>();
```

Reply:
Ah, both would be same.
But I used former one because concluding with semicolon is not recommended for macro.
(https://wiki.sei.cmu.edu/confluence/display/c/PRE11-C.+Do+not+conclude+macro+definitions+with+a+semicolon)

And maybe that's why following codes are wrapped with do-while statement :)
https://github.com/Samsung/ONE/blob/4c9870d3b781fe1f422e082d4bdee3b1a1638d0e/compiler/luci/logex/src/FormattedGraph.cpp#L2241-L2246

Reply:
OK, thanks for checking!

> semicolon is not recommended for macro.

We can use like this for this case.
```cxx
#define CIRCLE_NODE(OPCODE, CLASS)    \
  case luci::CircleOpcode::OPCODE: return std::make_shared<CLASS>()
```

I think I missed about `\` that this is actually make in a single line.